manager/controlapi/service.go

```diff
@@ -249,6 +299,12 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 			return errNetworkUpdateNotSupported
 		}
 
+		if request.Spec != nil && request.Spec.Endpoint != nil && (!reflect.DeepEqual(request.Spec.Endpoint, service.Endpoint) || !reflect.DeepEqual(request.Spec.Endpoint, service.Spec.Endpoint)) {
+			if err := s.checkPortConflicts(request.Spec); err != nil {
```

Comment:
Do we want to call this inside the `Update` transaction? See https://github.com/docker/swarmkit/pull/1188/files#r71379866